### PR TITLE
Fix combo usb creation on older versions of windows

### PR DESCRIPTION
**Pull request by ManMower**
Windows 7 and 8 combo USB stick creation currently fails because windows 7 and 8 have trouble
with multiple partitions on USB sticks. We work around this by partitioning the device in multiple
steps, so we only need to access the first partition at any time.

https://phabricator.endlessm.com/T29949

### CODE REVIEW

**Comment by ManMower** [comment]
Last minute single patch addition to fix an odd issue when cancelling.

**Comment by ManMower** [comment]
Just pushed a new trivial commit to fix running the installer from a combo stick.

This is going to be a violation of the test plan as soon as I update the test plan, so I figured I could be lazy and append that patch to this series instead of filing a new ticket.

File: endless/src/endless/EndlessUsbToolDlg.cpp [hunk 1]
```diff
@@ -4869,37 +4869,37 @@ DWORD WINAPI CEndlessUsbToolDlg::CreateUSBStick(LPVOID param)
...
 	/* We changed the partition table, so we need to get rufus to update its internal state so
 	 * stuff like FormatPartition and AltMountVolume can figure out volume names.
 	 * GetDrivePartitionData gets the physical lock itself, so we have to drop that first. */
 	safe_unlockclose(hPhysical);
+
+	// Waiting for offset zero just waits until any partition from the drive is present, so it
+	// can safely be called without getting partition data first. We do so to make sure the
+	// GetDrivePartitionData() call will see something.
+	uprintf("Waiting for logical drive to reappear...\n");
+	Sleep(200); // Radu: check if this is needed, that's what rufus does; I hate sync using sleep
```
Comment (wjt) [inline endless/src/endless/EndlessUsbToolDlg.cpp:4891]: Radu's name will always be present in this codebase.

File: endless/src/endless/EndlessUsbToolDlg.cpp [hunk 2]
```diff
@@ -5043,7 +5053,8 @@ bool CEndlessUsbToolDlg::SetupComboDisk(HANDLE hPhysical)
 error:
     return result;
 }
-bool CEndlessUsbToolDlg::CreateUSBPartitionLayout(HANDLE hPhysical)
+
+bool CEndlessUsbToolDlg::CreateUSBPartitionLayout(HANDLE hPhysical, bool just_esp)
```
Comment (wjt) [inline endless/src/endless/EndlessUsbToolDlg.cpp]: It's a minor thing and not worth revving this patch series for but I don't like boolean parameters. Perhaps:

```c++
enum CreatePartitionMode {
    JUST_ESP,
    EVERYTHING
}
```

**Review by wjt — APPROVED** [review]

**Comment by ManMower** [comment]
It's an easy enough change to use an enum, so I've made it.

Not much of a C++ programmer, I wasn't sure what scope to make the enum at, so I've made it at the same scope as other enum's in the same header - let me know if that's stupid.

**Comment by wjt** [comment]
I'm not much of a C++ programmer either :)

**Pull request merged**